src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -962,6 +1000,16 @@ public void stateChanged(ChangeEvent event) {
                     ((EntryEditorTab) activeTab).updateAll();
                     activateVisible();
                 }
+
+                // When the tab "Related articles" gets selected, the request to get the recommendations is started.
+                if (((JTabbedPane) event.getSource()).getSelectedComponent().getName()
+                        .equals(Localization.lang("Related articles"))) {
+                    if (relatedAcrticlesTabNeedsRecommendations) {
```

Comment:
Please move the logic related to `relatedAcrticlesTabNeedsRecommendations` to the `EntryEditorTabRelatedArticles` class. Here you just inform the tab that it got the focus.